extensions/wikia/ImageReview/ImageReviewHelper.class.php

```diff
@@ -194,6 +194,72 @@ public function refetchImageListByTimestamp( $timestamp ) {
 		return $imageList;
 	}
 
+	public static function checkImageValidity( $image ) {
+		$bValidImage = true;
+		$oImagePage = GlobalTitle::newFromId( $image->page_id, $image->wiki_id );
+		$oImageGlobalFile = null;
+		$sReason = 'unknown';
+
+		if ( $oImagePage instanceof GlobalTitle !== true ) {
+			$bValidImage = false;
+			$sReason = 'Page does not exist';
+		} elseif ( $oImagePage->isRedirect() === true ) {
+			$bValidImage = false;
+			$sReason = 'Page is a redirect';
+		} else {
+			$oImageGlobalFile = new GlobalFile( $oImagePage );
+			if ( $oImageGlobalFile->exists() === false ) {
+				$bValidImage = false;
+				$sReason = 'File does not exist';
+			}
+		}
+
+		if ( $bValidImage === true && $oImageGlobalFile instanceof GlobalFile ) {
```

Comment:
What do you think about moving `$oImageGlobalFile instanceof GlobalFile` check to if statement with `$oImageGlobalFile->exists() === false`?